.github/workflows/update-boringssl.yml

```diff
@@ -0,0 +1,222 @@
+  
+  workflow_dispatch:
+    inputs:
+      boringssl_revision:
+        description: 'Specific BoringSSL revision (SHA) to update to (leave empty for latest)'
+        required: false
+        type: string
+
+jobs:
+  update-boringssl:
+    runs-on: ubuntu-latest
+    permissions:
+      contents: write
+      pull-requests: write
+    
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+        with:
+          token: ${{ secrets.GITHUB_TOKEN }}
+          fetch-depth: 0
+
+      - name: Set up Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: '3.11'
+
+      - name: Set up Dart
+        uses: dart-lang/setup-dart@v1
+        with:
+          sdk: stable
+
+      - name: Set up Git
+        run: |
+          git config --global user.name 'github-actions[bot]'
+          git config --global user.email 'github-actions[bot]@users.noreply.github.com'
+
+      - name: Clone BoringSSL to get latest revision
+        id: get-revision
+        run: |
+          # Clone BoringSSL to a temporary directory to get the latest revision
+          TEMP_DIR=$(mktemp -d)
+          git clone https://boringssl.googlesource.com/boringssl "$TEMP_DIR/boringssl"
+          cd "$TEMP_DIR/boringssl"
+          
+          if [ -n "${{ github.event.inputs.boringssl_revision }}" ]; then
+            LATEST_REVISION="${{ github.event.inputs.boringssl_revision }}"
+            echo "Using manually specified revision: $LATEST_REVISION"
+          else
+            LATEST_REVISION=$(git rev-parse HEAD)
+            echo "Latest BoringSSL revision: $LATEST_REVISION"
+          fi
+          
+          # Get current revision from the Python script
+          CURRENT_REVISION=$(grep "BORINGSSL_REVISION = " tool/update-boringssl.py | cut -d"'" -f2)
+          echo "Current revision in script: $CURRENT_REVISION"
+          
+          echo "latest_revision=$LATEST_REVISION" >> $GITHUB_OUTPUT
+          echo "current_revision=$CURRENT_REVISION" >> $GITHUB_OUTPUT
+          
+          # Check if update is needed
+          if [ "$LATEST_REVISION" = "$CURRENT_REVISION" ]; then
+            echo "needs_update=false" >> $GITHUB_OUTPUT
+            echo "No update needed - already at latest revision"
+          else
+            echo "needs_update=true" >> $GITHUB_OUTPUT
+            echo "Update needed: $CURRENT_REVISION -> $LATEST_REVISION"
+          fi
+          
+          # Cleanup
+          rm -rf "$TEMP_DIR"
+
+      - name: Update BoringSSL revision in script
+        if: steps.get-revision.outputs.needs_update == 'true'
+        run: |
+          # Update the BORINGSSL_REVISION in the Python script
+          sed -i "s/BORINGSSL_REVISION = '[^']*'/BORINGSSL_REVISION = '${{ steps.get-revision.outputs.latest_revision }}'/" tool/update-boringssl.py
+          
+          # Verify the change
+          echo "Updated revision in script:"
+          grep "BORINGSSL_REVISION = " tool/update-boringssl.py
```

Comment:
Again why not make a bash script in `tool/bump-boringssl-revision.sh`

Or if we just store the revision in a `REVISION` file, then we don't need a script :D